### PR TITLE
Jest tests on UI

### PR DESCRIPTION
**Pull request by ylavoie**
This PR adds:
- Jest setup to test JavaScript code for the UI
- Reporting of said tests to Coveralls to document coverage
- Tests for LoginPage
Jest testing isolates the UI and mock all server interaction, allowing robust, repetitive and quick tests

### CODE REVIEW

**Comment by ehuelsmann** [comment]
@ylavoie is this really ready for merging? Or is this branch still waiting for Coveralls?

**Comment by ylavoie** [comment]
> @ylavoie is this really ready for merging? Or is this branch still waiting for Coveralls?



> @ylavoie is this really ready for merging? Or is this branch still waiting for Coveralls?

I was waiting. They answered 5 days ago, see https://github.com/lemurheavy/coveralls-public/issues/1662. I'll fix that.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -228,6 +228,8 @@ jobs:
           install-modules: |
             Devel::Cover
             Devel::Cover::Report::Coveralls
+            Syntax::Keyword::Try
```
[the remark this answers is not in the document]
Comment (ylavoie) [reply .github/workflows/main.yml]: Fixed

File: .circleci/config.yml [hunk 2]
```diff
@@ -11,6 +11,10 @@ aliases:
     store_artifacts:
       path: /tmp/artifact/
 
+# UI Coverage orbs
+orbs:
+  coveralls: coveralls/coveralls@1.0.6
```
[the remark this answers is not in the document]
Comment (ylavoie) [reply .circleci/config.yml:16]: Our CircleCI config still has provisions for coverage. This is for completeness, is case we want to enable it again somewhen

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -155,7 +155,7 @@ jobs:
       BROWSER: ${{ matrix.BROWSER }}
       COA_TESTING: ${{ matrix.COA_TESTING }}
       DB_TESTING: ${{ matrix.DB_TESTING }}
-      DEVEL_COVER_OPTIONS: -silent,1,+ignore,(^x?t/|^local/|^utils/|\.lttc$|^/usr/|^/opt/|starman\$\$),-summary,1
+      DEVEL_COVER_OPTIONS: -silent,1,+ignore,(^x?t/|^local/|^utils/|\.lttc\$|^/usr/|^/opt/|starman\$\$),-summary,1
```
[the remark this answers is not in the document]
Comment (ylavoie) [reply .github/workflows/main.yml]: Otherwise `$,` was interpreted. I changed it completely to simplify and make it more readable

**Comment by ylavoie** [comment]
> @ylavoie is this really ready for merging? Or is this branch still waiting for Coveralls?

Seems ok now

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -125,16 +125,19 @@ jobs:
             postgres: "13"
             BROWSER: "chrome"
             COVERAGE: 1
+            JOB_COUNT: 2
```
Comment (ehuelsmann) [inline .github/workflows/main.yml:128]: This changes the build matrix, which means adjustment is required for the required jobs on branch protection...

**Review by ehuelsmann — COMMENTED** [review]
.

**Pull request merged**